### PR TITLE
HTCONDOR-1061 ARC CE token authentication

### PR DESCRIPTION
**Pull request by JaimeFrey**
Support token authentication to ARC CE servers.

# HTCondor Pull Request Checklist for internal reviewers

- [x] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [x] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [x] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [x] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [x] Verify that the branch destination of the PR matches the target version of the ticket
- [x] Check for correctness of change
- [x] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [x] Check for documentation, if needed
- [x] Check for version history, if needed
- [x] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [x] Check that commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

File: src/condor_gridmanager/arcresource.cpp [hunk 1]
```diff
@@ -142,16 +154,18 @@ void ArcResource::DoPing( unsigned& ping_delay, bool& ping_complete,
 {
 	int rc;
 
-	if ( gahp->isInitialized() == false ) {
+	if ( gahp->isStarted() == false ) {
```
Comment (Todd-L-Miller) [inline src/condor_gridmanager/arcresource.cpp:157]: This change seems unrelated to the rest?

**Review by Todd-L-Miller — APPROVED** [review]
At some point we should move all our copies of `readShortFile()` to some sort of common source/library, but that's for later.

**Comment by Todd-L-Miller** [comment]
There's not practical way (AFAIK) to test ARC submission at all yet, so there's no need to test token-based submission specifically.

File: src/condor_gridmanager/arcresource.cpp (hunk 1 above)
Comment (JaimeFrey) [reply src/condor_gridmanager/arcresource.cpp:157]: isInitialized() returns false if a proxy hasn't be set for the gahp_server. With token auth, there is no proxy, so we just want to wait until the gahp is running. We set the proxy immediately after gahp startup (in ArcJob), so it's okay to use isStarted() in all cases.
